src/zenml/utils/enum_utils.py

```diff
@@ -0,0 +1,30 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from enum import Enum
+from typing import List, cast
+
+
+class StrEnum(str, Enum):
+    """Base enum type for string enum values"""
+
+    def __str__(self) -> str:
+        """Returns the enum string value."""
+        return self.value  # type: ignore
+
+    @classmethod
+    def list(cls) -> List[str]:
+        """Get all enum names as a list of strings"""
+        names = map(lambda c: c.name, cls)  # type: ignore
```

Comment:
I'm a little curious about this kind of roundabout approach of using a map, then casting to list using `typing` module and then casting to list with the python builtin? Would it not also work to just define this function as:
```return [c.name for c in cls]```
?